libs/language-server/capabilities/document-links/src/lib/target-link.ts

```diff
@@ -41,20 +48,39 @@ export async function targetLink(
       throw 'local @nx/devkit dependency not found';
     }
 
+    const nxVersion = await getNxVersion(workingPath);
+
     const importPath = join(devkitPath, 'src/executors/parse-target-string');
     const { parseTargetString } = await importWorkspaceDependency<
       typeof import('@nx/devkit/src/executors/parse-target-string')
     >(importPath, lspLogger);
-    const parsedTargets = parseTargetString(targetString);
-    project = parsedTargets.project;
-    target = parsedTargets.target;
-    configuration = parsedTargets.configuration;
+    let parsedTarget: Target;
+    if (gte(nxVersion.full, '17.0.6')) {
```

Comment:
`semver.gte` will throw if `nxVersion.full` is not a valid `semver` string. 

Do we want to add a `try-catch` around this to prevent crashes and have some kind of fallback? It might never hit it, I'm not sure the order of operations here, but a thought at least.

Reply:
it's a valid point, I'll leave it for now but I have a refactor planned sometime in the future for the whole nxVersion stuff. We need to break out from semver by treating `0.0.0-pr` versions as `999` or smth otherwise nx console is just broken on pr releases